From 9e4092102042d3d848455e13be00ac29048ad56d Mon Sep 17 00:00:00 2001
From: vipinc <vipinc@ami.com>
Date: Wed, 18 Oct 2023 11:50:11 +0530
Subject: [PATCH] EMail: Update the install location for systemd service

This patch update the install location for systemd service and
update the C++ version to 20

Signed-off-by: vipinc <vipinc@ami.com>
---
 CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cab24e5..cb56366 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
 project(mail-alert-manager CXX)
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
@@ -22,6 +22,8 @@ pkg_check_modules(SYSTEMD libsystemd REQUIRED)
 # import sdbusplus
 find_package(PkgConfig REQUIRED)
 pkg_check_modules(SDBUSPLUSPLUS sdbusplus REQUIRED)
+pkg_get_variable(SYSTEMD_SYSTEM_UNIT_DIR systemd systemdsystemunitdir)
+
 
 add_executable(mail-alert-manager ${SRC_FILES})
 
@@ -37,5 +39,5 @@ link_directories(${EXTERNAL_INSTALL_LOCATION}/lib)
 set(SERVICE_FILES
     ${PROJECT_SOURCE_DIR}/service_files/mail-alert-manager.service)
 
-install(FILES ${SERVICE_FILES} DESTINATION /lib/systemd/system/)
+install(FILES ${SERVICE_FILES} DESTINATION ${SYSTEMD_SYSTEM_UNIT_DIR})
 install(TARGETS mail-alert-manager DESTINATION bin)
-- 
2.25.1

